### PR TITLE
CO Renaming (Pt. 5): Latency Control

### PR DESCRIPTION
**Pull request by Holzhaus**
*Part of https://github.com/mixxxdj/mixxx/issues/11931.*

Moves the `latency` control into the `[App]` group.

### CODE REVIEW

**Comment by JoergAtGithub** [comment]
It would be future proof to call it `output_latency` instead of just `latency`. We currently do not use the input-latency value that portaudio reports, but for more complex multi device sync scenarios it might be needed too.

**Comment by Holzhaus** [comment]
> It would be future proof to call it `output_latency` instead of just `latency`. We currently do not use the input-latency value that portaudio reports, but for more complex multi device sync scenarios it might be needed too.

@daschuer @Swiftb0y opinions?

File: src/engine/enginemixer.cpp [hunk 1]
```diff
@@ -79,9 +79,10 @@ EngineMixer::EngineMixer(
     m_pMainSampleRate->set(44100.);
 
     // Latency control
-    m_pMainLatency = new ControlObject(ConfigKey(group, "latency"),
+    m_pMainLatency = new ControlObject(ConfigKey(kAppGroup, QStringLiteral("latency")),
```
[the remark this answers is not in the document]
Comment (daschuer) [reply src/engine/enginemixer.cpp:82]: Or the new name you decide for ...

**Review by daschuer — CHANGES_REQUESTED** [review]

**Comment by Holzhaus** [comment]
I opted for `output_latency_ms`.

File: src/test/co_dumps/co_dump_inital.csv [hunk 2]
```diff
@@ -29,7 +29,7 @@
 [EffectRack1_EffectUnit4_Effect4],parameter4_set_default,0
 [EffectRack1_EffectUnit2_Effect2],parameter2_link_type,0
 [QuickEffectRack1_[Channel3]_Effect1],parameter5_set_zero,0
-[Master],latency,5.80499
+[App],output_latency_ms,5.80499
```
Comment (daschuer) [inline src/test/co_dumps/co_dump_inital.csv:32]: My dump button is broken. Does yours work? I will file a bug.

**Review by daschuer — CHANGES_REQUESTED** [review]
I think the Akai mapping is still broken. The reported output_latency can be long and is unrelated to the internal processing. 
I have found out where the "audio_buffer_size" CO is gone that is probably the correct use here: 
https://github.com/mixxxdj/mixxx/pull/11121/files

But does the script really need that compensation? Why do we not see the use case in other mappings? 

Did you consider how to continue?

File: src/test/co_dumps/co_dump_inital.csv (hunk 2 above)
Comment (Holzhaus) [reply src/test/co_dumps/co_dump_inital.csv:32]: > My dump button is broken. Does yours work? I will file a bug. 

Last time it check it didit creates a file in `~/.mixxx`. It does not create a save file dialog.

Comment (daschuer) [reply src/test/co_dumps/co_dump_inital.csv:32]: Oh, right it is working. I just needed to look in my testing .mixxx folder 🤦

**Comment by Holzhaus** [comment]
> I think the Akai mapping is still broken. The reported output_latency can be long and is unrelated to the internal processing. 
> I have found out where the "audio_buffer_size" CO is gone that is probably the correct use here: 
> https://github.com/mixxxdj/mixxx/pull/11121/files
> 
> But does the script really need that compensation? Why do we not see the use case in other mappings? 
> 
> Did you consider how to continue? 

Yes, I considered it and decided that I won't mess with the mapping in this PR. I updated the control, but I don't plan other changes. The bug was not caused by this PR and I don't own this controller, so I cannot properly test changes.

**Comment by daschuer** [comment]
OK, what puzzles me is the question if we need to revive the "audio_buffer_size" for such use cases. The bug in the Akai mapping itself is there or not, probably hardly notable. Do you have there a good idea for a bug report regarding the that?

**Comment by Holzhaus** [comment]
> OK, what puzzles me is the question if we need to revive the "audio_buffer_size" for such use cases.

The use case is "implementing backjumps in JavaScript instead of using the proper Mixxx COs" ?

**Comment by daschuer** [comment]
> The use case is "implementing beatjumps in JavaScript instead of using the proper Mixxx COs" ?

Ok, that is clearly not a valid use case. So there is nothing else to do.  
Thank you for this PR.

**Review by daschuer — APPROVED** [review]

**Comment by daschuer** [comment]
We can also ignore the Lint issues. Someone really needs to put some love on this mapping.

**Pull request merged**